api/src/utils/requestId.js

```diff
@@ -0,0 +1,12 @@
+import { v4 as uuidv4 } from 'uuid';
+import { session } from './requestContext.js';
+
+export const requestIdMiddleware = (_req, _res, next) => {
+  session.run(() => {
+    const id = uuidv4();
+    session.set('requestId', id);
+    next();
+  });
+};
+
+export const getRequestId = () => session.get('requestId');
```

Comment:
`getRequestId()` may throw if executed outside of the callback chain, I would try/catch and add a fallback to avoid throwing 
```suggestion
export const getRequestId = () => {
  try {
    // may throw if executed outside of the callback chain
    return session.get('requestId');
  } catch (e) {
    return undefined;
  }
};
```

Reply:
I'm not sure this can throw any error, it would just return `undefined` if `requestId` is not present in the session.
But I'm not 100% sure so I'm fine adding this `try / catch` 👍